src/main/kotlin/io/spinnaker/bumpdeps/Main.kt

```diff
@@ -81,6 +89,36 @@ class BumpDeps : CliktCommand() {
         }
     }
 
+    private fun waitForArtifact() {
+        if (mavenRepositoryUrl.isNullOrBlank() || groupId.isNullOrBlank() || artifactId.isNullOrBlank()) {
+            Thread.sleep(Duration.ofMinutes(10).toMillis())
+            return
+        }
+
+        val groupPath = groupId!!.replace('.', '/')
+        val okHttpClient = OkHttpClient.Builder().build()
+        val request = Request.Builder()
+            .url("$mavenRepositoryUrl/$groupPath/$artifactId/$version/$artifactId-$version.pom")
+            .build()
+
+        val start = Instant.now()
+        val retryDelay = Duration.ofSeconds(30)
+        val maxTotalDelay = Duration.ofMinutes(10)
+        while (true) {
+            val response = okHttpClient.newCall(request).execute()
+            if (response.isSuccessful) {
+                return
```

Comment:
might be interesting to log something here to show how long it took for the artifact to become available if there isn't already some sort of logging around the start / completion of this bit

Reply:
Great idea; done